components/base/wheeled/kinematics.go

```diff
@@ -51,13 +61,74 @@ func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
 }
 
 func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
-	// TODO(RSDK-2311): complete the implementation
-	return []referenceframe.Input{}, errors.New("not implemented yet")
+	// TODO(rb): make a transformation from the component reference to the base frame
+	pose, _, err := kwb.slam.GetPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pt := pose.Point()
+	theta := math.Mod(pose.Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	return []referenceframe.Input{{Value: pt.X}, {Value: pt.Y}, {Value: theta}}, nil
 }
 
-func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// TODO(RSDK-2311): complete the implementation
-	return errors.New("not implemented yet")
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, inputs []referenceframe.Input) (err error) {
+	// create a frame system to be used during the context of this function, used to locate the goal in the base frame
+	fs := referenceframe.NewEmptyFrameSystem("")
+	if err := fs.AddFrame(kwb.model, fs.World()); err != nil {
+		return err
+	}
+
+	// create a function for the error state, which is defined as [positional error, heading error]
+	errorState := func() (int, float64, error) {
+		currentInputs, err := kwb.CurrentInputs(ctx)
+		if err != nil {
+			return 0, 0, err
+		}
+
+		// create a goal pose in the world frame
+		desiredHeading := math.Atan2(currentInputs[1].Value-inputs[1].Value, currentInputs[0].Value-inputs[0].Value)
+		goal := referenceframe.NewPoseInFrame(
+			referenceframe.World,
+			spatialmath.NewPose(
+				r3.Vector{X: inputs[0].Value, Y: inputs[1].Value},
+				&spatialmath.OrientationVector{OZ: 1, Theta: desiredHeading},
+			),
+		)
+
+		// transform the goal pose such that it is in the base frame
+		tf, err := fs.Transform(map[string][]referenceframe.Input{kwb.name: currentInputs}, goal, kwb.name)
+		if err != nil {
+			return 0, 0, err
+		}
+		delta, ok := tf.(*referenceframe.PoseInFrame)
+		if !ok {
+			return 0, 0, errors.New("can't interpret transformable as a pose in frame")
+		}
+
+		// calculate the error state
+		headingErr := math.Mod(delta.Pose().Orientation().OrientationVectorDegrees().Theta, 360)
+		positionErr := int(1000 * delta.Pose().Point().Norm())
+		kwb.logger.Debugf("Position error: \t%d mm\tHeading error: \t%f degrees", positionErr, headingErr)
+		return positionErr, headingErr, nil
+	}
+
+	// this loop polls the error state and issues a corresponding command to move the base to the objective
+	// when the base is within the positional threshold of the goal, exit the loop
+	// TODO(rb): check for the context being cancelled and stop if so
+	for distErr, headingErr, err := errorState(); err == nil && distErr > positionThresholdMM; distErr, headingErr, err = errorState() {
```

Comment:
I think this might be missing these error checks from the tech spec:
- If distance from the straight line path between the start and the goal inputs exceeds some default “acceptable” amount, exit and return error
- If time since start exceeds 5 minutes and GetPositon hasn’t changed within a specific range in that duration then exit and return error.

Reply:
sounds good!